### DISCUSSION TITLE
Suggestions for nex-gen deepin desktop

### DISCUSSION THREAD

**Discussion by raspin0** [linuxdeepin/developer-center#3258] [Features Request & Ideas | 特性请求 & 头脑风暴]
As far as I know there are plans to create a next gen desktop, so I'd want to share my suggestions. 

- Window preview under alt+tab 
- Adapt folder icons with accent colour
- Close file preview(space) in File Manager after losing focus + add "always on top"
- Speed up repeat delay on Function keys(Fn) e.g. while changing sound volume
- Privacy indicators when mic, camera or screen capture is used
- Add small launcher animation
- Add new system sounds that are more gentle to ears no matter what volume level

I hope those suggestions will be helpful, and thank you for making this awesome desktop!

**Comment by BLumia** [comment]
> Adapt folder icons with accent colour

This could be not hard to implement but I'm not sure if there are any plans to support it. @meiyixiang 

@justforlxz As currently deepin's icon engine is somewhat broken under Arch (and probably other distros), maybe we can consider add an (advanced) setting option to make it load other icon engines? This will also be a way to support accent color on icons before we introduce such support in our own icon engine.

**Reply by raspin0** [reply]
Another bunch of guidelines from my end after experiencing deepin desktop
1. Deepin should pre-install and support emoji font in color. At the top there are google's noto emoji and twitter twemoji. I find twitter font much prettier, but both are open-source and free to use even in comercial products https://twemoji.twitter.com/
b) There should be an option to change emoji font from the Control center.
2. Add any indicator the File manager when running as administrator to let the user know they are operating with elevated privileges.
3. If Trash is already open in File manager and it goes beneath other windows, put it on focus again after clicking the icon on the dock
4. Add close animation. I find windows/gnome animations too slow, but using an animation with a speed of 3-4ms it's possible to add a very nice feeling to closing windows.
5. Improve handling multiple windows from dock. The current form is very confusing and hard to use. It's confusing because I don't know what is currently open and hovering over the icon is extremely slow. The windows behaviour is in my opinion better, because click action pulls up windows you want to use.
